src/ts/ConvivaAnalytics.ts

```diff
@@ -71,20 +87,14 @@ export class ConvivaAnalytics {
       return; // Cancel initialization
     }
 
-    // player versions <=7.2 did not have a ON_PLAYING event
-    // we track this change to correctly transition to the playing state
-    this.hasPlayingEvent = Boolean(player.EVENT.ON_PLAYING);
     this.sessionDataPopulated = false;
 
-    // Assert that this class is instantiated before player.setup() is called.
-    // When instantiated later, we cannot detect startup error events because they are fired during setup.
-    if (player.isReady()) {
```

Comment:
Please check if there is already a source loaded instead.

```ts
if (!player.getSource()) {
   console.error('ConvivaAnalytics must be instantiated before calling player.load()');
  return; // Cancel initialization
}
```

Reply:
Should be `if (player.getSource())` w/o `!`, right?

Reply:
true sry.